types/hashes.go

```diff
@@ -0,0 +1,37 @@
+package types
+
+import "hash"
+
+// GenericDigest is a digest that implements CustomSizeDigest, and can be used as hash output.
+//msgp:allocbound GenericDigest MaxHashDigestSize
+type GenericDigest []byte
+
+// HashID is a domain separation prefix for an object type that might be hashed
+// This ensures, for example, the hash of a transaction will never collide with the hash of a vote
+type HashID string
+
+// Hashable is an interface implemented by an object that can be represented
+// with a sequence of bytes to be hashed or signed, together with a type ID
+// to distinguish different types of objects.
+type Hashable interface {
```

Comment:
This generic type is very useful on go-algorand, but I think we can use a simple version here. i.e let's just create a function that hashes the stateproof message and the light block header.